pkgs/top-level/python-packages.nix

```diff
@@ -27805,6 +27805,36 @@ in {
     };
   };
 
+  worldengine = buildPythonPackage rec {
+    name = "worldengine-${version}";
+    version = "0.19.0";
+
+    src = pkgs.fetchurl {
+      url = "mirror://pypi/w/worldengine/${name}.tar.gz";
+      sha256 = "0wnpp0zm4idc4242a970yf1blf8y2f1ikwha0q781wfz1dg4z0i6";
+    };
+
+    propagatedBuildInputs = with self; [ noise numpy pyplatec protobuf3_1 purepng argparse h5py gdal ];
+    doCheck = true;
+
+    prePatch = ''
+      substituteInPlace setup.py \
+        --replace pypng>=0.0.18 purepng \
+        --replace 'numpy>=1.9.2, <= 1.10.0.post2' 'numpy' \
+        --replace 'argparse==1.2.1' 'argparse' \
+        --replace 'protobuf==3.0.0a3' 'protobuf' \
+    '';
+      # --replace 'noise==1.2.2' 'noise' \
+      # --replace 'PyPlatec==1.4.0' 'PyPlatec' \
```

Comment:
I wrote the code to drop all specific version, but those 2 are actually the versions on nixpkgs, so I commented them out. I guess it's pointless now, will remove.